src/plots/gl3d/scene.js

```diff
@@ -790,6 +790,9 @@ proto.plot = function(sceneData, fullLayout, layout) {
             y: fullSceneLayout.aspectratio.y,
             z: fullSceneLayout.aspectratio.z
         };
+
+        // also keep track of aspectmode here
+        scene.viewInitial.aspectmode = fullSceneLayout.aspectmode;
```

Comment:
Shouldn't this be

```js
if(!scene.viewInitial.aspectmode) {
  scene.viewInitial.aspectmode = fullSceneLayout.aspectmode
}
```

that way we don't override `viewInitial` on updates?

Reply:
Good call.
We could. Done in https://github.com/plotly/plotly.js/pull/4579/commits/c730f1276ad167adf44b13cf8482fb317d88843c.